FastRoute.hhi

```diff
@@ -38,7 +38,12 @@ namespace FastRoute {
           'dataGenerator' => ?classname<DataGenerator>,
           'dispatcher' => ?classname<Dispatcher>,
           'routeCollector' => ?classname<RouteCollector>,
-        ) $options = shape()): Dispatcher;
+        ) $options = shape(
+          'routeParser' => \FastRoute\RouteParser\Std::class,
+          'dataGenerator' => \FastRoute\DataGenerator\GroupCountBased::class,
+          'dispatcher' => \FastRoute\Dispatcher\GroupCountBased::class,
+          'routeCollector' => \FastRoute\RouteCollector::class
+        )): Dispatcher;
```

Comment:
Wouldn't it be better to mark the keys as optional (rather than values as nullable)? As I understand it, after your fix here this implementation would still require specifying all keys explicitly, right?

I realize that would not be compatible with old Hack versions, but I think it's better to support the current one correctly.

Reply:
Alright, I changed it so that these fields are optional instead of adopting the nullable types.